src/rust/engine/src/lib.rs

```diff
@@ -546,6 +547,31 @@ pub extern fn rule_subgraph_visualize(
   })
 }
 
+#[no_mangle]
+pub extern fn set_panic_handler() {
+  panic::set_hook(Box::new(|panic_info| {
+    let mut panic_str = format!("panic at '{}', ",
+                                panic_info.payload().downcast_ref::<&str>().unwrap());
+
+    if let Some(location) = panic_info.location() {
+      let panic_location_str = format!("{}:{}", location.file(), location.line());
+      panic_str.push_str(&panic_location_str);
+    }
+
+    externs::log(externs::LogLevel::Critical, &panic_str);
+
+    let mut panic_file_bug_str = "Oops! You should not see this. ".to_string();
+    panic_file_bug_str.push_str("Please file a bug at https://github.com/pantsbuild/pants/issues.");
```

Comment:
would kill the "oops you should not see this" part in favor of just the "Please file a bug at ..." string.